clusterloader2/pkg/test/test.go

```diff
@@ -37,41 +38,77 @@ var (
 	Test = createSimpleExecutor()
 )
 
-// RunTest runs test based on provided test configuration.
-func RunTest(
+// CreateTestContext creates the test context.
+func CreateTestContext(
 	clusterFramework *framework.Framework,
 	prometheusFramework *framework.Framework,
 	clusterLoaderConfig *config.ClusterLoaderConfig,
 	testReporter Reporter,
-) *errors.ErrorList {
+	testScenario *api.TestScenario,
+) (Context, *errors.ErrorList) {
 	if clusterFramework == nil {
-		return errors.NewErrorList(fmt.Errorf("framework must be provided"))
+		return nil, errors.NewErrorList(fmt.Errorf("framework must be provided"))
 	}
 	if clusterLoaderConfig == nil {
-		return errors.NewErrorList(fmt.Errorf("cluster loader config must be provided"))
+		return nil, errors.NewErrorList(fmt.Errorf("cluster loader config must be provided"))
 	}
 	if CreateContext == nil {
-		return errors.NewErrorList(fmt.Errorf("no CreateContext function installed"))
+		return nil, errors.NewErrorList(fmt.Errorf("no CreateContext function installed"))
+	}
+
+	mapping, errList := config.GetMapping(clusterLoaderConfig, testScenario.OverridePaths)
+	if errList != nil {
+		return nil, errList
 	}
+
+	return CreateContext(clusterLoaderConfig, clusterFramework, prometheusFramework, state.NewState(), testReporter, mapping, testScenario), errors.NewErrorList()
+}
+
+// CompileTestConfig loads the test configuration and nested modules.
+func CompileTestConfig(ctx Context) (*api.Config, *errors.ErrorList) {
```

Comment:
Let's make it return error instead of errors.ErrorList. There is only one caller of this method and it converts the returned errorList to a single error anyway.

Reply:
The error list could be returned by `testConfig.Validate()`, now it's a list of one because there is only one check, but potentially longer after the config validation PR.

Did you mean the `CreateTestContext` function instead? It does return single errors, I didn't see the use of errList.Append.